include/umf/providers/provider_cuda.h

```diff
@@ -53,6 +53,22 @@ umf_result_t umfCUDAMemoryProviderParamsSetMemoryType(
     umf_cuda_memory_provider_params_handle_t hParams,
     umf_usm_memory_type_t memoryType);
 
+/// @brief  Set the host allocation flags in the parameters struct.
+/// @param  hParams handle to the parameters of the CUDA Memory Provider.
+/// @param  flags combination of CU_MEMHOSTALLOC_PORTABLE,
+///         CU_MEMHOSTALLOC_DEVICEMAP and CU_MEMHOSTALLOC_WRITECOMBINED flags
+/// @return UMF_RESULT_SUCCESS on success or appropriate error code on failure.
+umf_result_t umfCUDAMemoryProviderParamsSetHostAllocFlags(
```

Comment:
Why do we need 2 separate functions? A provider can allocate either host or managed memory, never both, right?
If It's only for checking if the flags are valid, we can move validation to initialize.

Reply:
merged, done